lib/embulk/output/bigquery/file_writer.rb

```diff
@@ -31,20 +31,35 @@ def initialize(task, schema, index, converters = nil)
             end
           end
 
-          @path = sprintf("#{@task['path_prefix']}#{@task['sequence_format']}#{@task['file_ext']}", Process.pid, index)
-          Embulk.logger.info { "embulk-output-bigquery: will create #{@path}" }
-          if File.exist?(@path)
-            Embulk.logger.warn { "embulk-output-bigquery: unlink already existing #{@path}" }
-            File.unlink(@path) rescue nil
+          @io = create_io
+          @path = @io.path
+        end
+
+        THREAD_LOCAL_IO_KEY = :embulk_output_bigquery_file_writer_io
+
+        # Create one io object for one thread, that is, share among tasks
+        # Close theses shared io objects in transaction
```

Comment:
I am requesting to add new API to get the number of `max_threads` at https://github.com/embulk/embulk/issues/412, but currently it is not available (in my understanging).

That is why I am creating new io objects **dynamically** at here.